File: KotlinBase/src/main/java/com/simon/java/algorithm/Recursive.java
package com.simon.java.algorithm;


public class Recursive {

    /**
     * 递归实现n的阶乘
     */
    public static int doRecursive(int n) {
        if (n == 0 || n == 1) {
            return 1;
        }
        return n * doRecursive(n - 1);
    }

}

File: KotlinBase/src/main/kotlin/com/simon/kotlin/algorithm/Recursive.kt
package com.simon.kotlin.algorithm

fun main(){

    println(foo(0))
    println(foo(1))
    println(foo(2))
    println(foo(3))
}

fun foo( n : Int): Int {
    return if (n == 0) 1 else n * foo(n - 1)
}